### PR TITLE
Add a new packaging pipeline

### PR DESCRIPTION
**Pull request by alexsapran**
## What does this PR do?
This is an empty initial pipeline that is going to be packaging elastic-agent.

## Why is it important?

This pipeline is intended to be used for packaging elastic-agent.

## Related issues

related: https://github.com/elastic/elastic-agent/issues/2595

### CODE REVIEW

File: catalog-info.yaml [hunk 1]
```diff
@@ -55,3 +55,45 @@ spec:
...
+      name: elastic-agent-package
+      description: Buildkite pipeline for packaging Elastic Agent package
+    spec:
+      repository: elastic/elastic-agent
+      pipeline_file: ".buildkite/pipeline.elastic-agent-package.yml"
+      provider_settings:
+        build_pull_request_forks: false
+        # 
+        build_pull_requests: true # requires filter_enabled and filter_condition settings as below when used with buildkite-pr-bot
+        build_tags: true
```
[the remark this answers is not in the document]
Comment (mrodm) [reply catalog-info.yaml:86]: Should it be disabled tags and branches for now ? 
```suggestion
        build_tags: false
        build_branches: false
```

File: catalog-info.yaml [hunk 2]
```diff
@@ -55,3 +55,45 @@ spec:
...
+    metadata:
+      name: elastic-agent-package
+      description: Buildkite pipeline for packaging Elastic Agent package
+    spec:
+      repository: elastic/elastic-agent
+      pipeline_file: ".buildkite/pipeline.elastic-agent-package.yml"
+      provider_settings:
+        build_pull_request_forks: false
+        # 
+        build_pull_requests: true # requires filter_enabled and filter_condition settings as below when used with buildkite-pr-bot
```
Comment (mrodm) [inline catalog-info.yaml:85]: As in the bot config `set_commit_status` is false
Should it be disabled here the status fo rcommits ? 
```suggestion
        build_pull_requests: true # requires filter_enabled and filter_condition settings as below when used with buildkite-pr-bot
        publish_commit_status: false # do not update status of commits for this pipeline
```

File: catalog-info.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexsapran) [reply catalog-info.yaml:87]: And we keep only the pr_bot comment trigger. I guess we would want to keep the tag, but once we are ready.

**Review by mrodm — DISMISSED** [review]
LGTM

**Review by mrodm — APPROVED** [review]

**Review by jlind23 — APPROVED** [review]

**Pull request merged**